### PR TITLE
Allow large user mapping range with fakeroot

### LINKED ISSUE #3957: unit test failure: FAIL: TestGetIDRangePath
This was seen on opensuse leap 15.0 while testing another PR. Please let me know if you have an idea to try or need more information. Thanks!

Information about the system:
```
$ uname -a
Linux vagrant-openSUSE-Leap 4.12.14-lp150.12.67-default #1 SMP Tue Jul 9 08:18:26 UTC 2019 (b610084) x86_64 x86_64 x86_64 GNU/Linux
```

```
=== RUN   TestGetIDRangePath
--- FAIL: TestGetIDRangePath (0.00s)
    fakeroot_test.go:37: unexpected error for "temporary file, uid 1 (multiple good)": could not retrieve user with UID 1: user: unknown userid 1
FAIL
coverage: 75.9% of statements
FAIL    github.com/sylabs/singularity/internal/pkg/fakeroot     0.006s
```

### LINKED ISSUE #4512: fakeroot uid/gid mapping length fixed to insufficient 65536 (with subuid/subgid).
### Version of Singularity:

```
3.4.1-1.el7
```

### Expected behavior

Expectation is 
```
$ singularity build --fakeroot /tmp/test.sifj test_galaxy.def
```
when run by `pontusf` on a host with 

```
$ cat /etc/subuid 
pontusf:10000000:4000000
$ cat /etc/subgid
pontusf:10000000:4000000
$ 
```
should build with fakeroot.

### Actual behavior

Build dies with error

```
FATAL:   could not use fakeroot: entries for user pontusf found in /etc/subuid but all with a range count different from 65536         
```
I can see valid reasons for enforcing a minimum mapping length to protect users from themselves, but no valid reasons for enforcing an upper limit assuming (excluding cases that would overflow a uid_t).

### Steps to reproduce this behavior

Set up a subgid/subuid file for the user running singularity with a mapping length other than 65536.

### What OS/distro are you running

```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

### How did you install Singularity

Custom rpms built from source.

### Use case motivation

We run with a somewhat "sparse" user and group space where in addition to system and administrative uids/gids (that are low), we use ranges close to 10 000 and 40 000-50 000 that have been grandfathered in.

Newer uid/gids will use values close to 4 000 000 (for uids/groups tied to users), 3 000 000 for project-related groups and 5 000 000 for a few other groups.

For us to be comfortable with using fakeroot, we'd need the mapping to cover all "reasonable" uids/gids (at least 6 000 000 with the uid/gid space we use today).

### PR DESCRIPTION
**Pull request by cclerget**
## Description of the Pull Request (PR):

Fakeroot feature read `/etc/subuid` and `/etc/subgid` files to determine user mappings, before this PR only entries with range count of 65536 were considered as valid, after this PR range count >= 65536 are valid (see #4512).
Also fixes unit tests for issue #3957 by mocking `GetPWUID` function

### This fixes or addresses the following GitHub issues:

 - Fixes #4512
 - Fixes #3957

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by sashayakovtseva — APPROVED** [review]

**Comment by DrDaveD** [comment]
@cclerget 
I tried this PR.  I first added myself with `sudo singularity config fakeroot --add dwd` and then added a second user.  Then I hand-edited /etc/sub[ug]id to have one line for me with the lower bound of the second user and a size of 2*65536, and then added the second user again.  So I ended up with /etc/subuid looking like this:
```
3382:4294770688:131072
994:4294705152:65536
```
where I am 3382.  I then tried `sudo singularity config fakeroot --remove dwd` and it said
```
FATAL:   while removing dwd: no mapping entry found in /etc/subuid for dwd
```
If I hand edit again my entry to be of range 65536, then --remove works.

So bottom line I guess, `config fakeroot --remove` does not seem to work unless the range is 65536.

**Comment by dtrudg** [comment]
@cclerget - can we mop up the issue raised by @DrDaveD here before this is merged? Or can you indicate if you feel this is a separate issue?

**Comment by cclerget** [comment]
@DrDaveD It's intentional to left `config` command handle only entries with a 65536 range as `config fakeroot --add` always add entry with a 65536 range, all entries with a range different from 65536 are considered as set externally

**Comment by DrDaveD** [comment]
@cclerget I don't see why config fakeroot --remove can't still work without the expected range, but in any case the error message is misleading; I think it should at least better explain why it refuses to remove.

**Comment by cclerget** [comment]
@DrDaveD I applied few modifications, could you check ?

**Comment by DrDaveD** [comment]
@cclerget Now I can remove a user that has a non-standard range, looks good.  Adding after that works also.
